### PR TITLE
v1.4: Changed loadQuestions() and user's answer sanity check for quiz

### PR DESCRIPTION
**Pull request by claysmilesoil**
- Changed loading questions from text file so that it generates the question numbers before loading the .txt files

- Add ability to filter user's answers during quiz. Does not accept the user's answer if it is not among the question's options.

### CODE REVIEW

File: src/main/java/javacake/quiz/QuestionList.java [hunk 1]
```diff
@@ -46,53 +47,51 @@ public String getAnswers(int index) {
...
                 while ((sentenceRead = reader.readLine()) != null) {
                     stringBuilder.append(sentenceRead).append("\n");
                 }
                 reader.close();
-                String[] questions = stringBuilder.toString().substring(0,stringBuilder.length() - 1).split("\\|\\s*");
-                this.tempList.add(new Question(questions[0], questions[1]));
             } catch (Exception e) {
                 throw new CakeException("Error in loading file :(");
             }
+            String[] questions = stringBuilder.toString().substring(0,stringBuilder.length() - 1).split("\\|\\s*");
```
Comment (GlenWong97) [inline src/main/java/javacake/quiz/QuestionList.java]: This line can be improvedto increase readability.

Comment (GlenWong97) [inline src/main/java/javacake/quiz/QuestionList.java]: This line can be expanded to improve readability, right now it is quite hard to comprehend.

File: src/main/java/javacake/quiz/Question.java [hunk 2]
```diff
@@ -31,7 +45,11 @@ public String getUserAnswer() {
      * @param input user's inputted answer
      * @return true if input matches answer, false otherwise.
      */
-    public boolean isAnswerCorrect(String input) {
+    public boolean isAnswerCorrect(String input) throws CakeException {
+        int tmp = Integer.parseInt(input);
+        if (tmp < 1 || tmp > numberOfOptions) {
```
Comment (GlenWong97) [inline src/main/java/javacake/quiz/Question.java]: **Suggestion**: This line can be refactored by using a boolean method to improve on readability.

E.g. `if (isWithinNumberOfOptions(int tmp)) {` 

and `private boolean isWithinNumberOfOptions(int tmp) {
return (tmp < 1 || tmp > numberOfOptions);
}`

**Review by RusdiHaizim — APPROVED** [review]
Good job on refactoring code to include number of options in each question!

**Review by GlenWong97 — APPROVED** [review]
Great implementation, however some parts of the code can be further improved. Do consider some of the suggestions made by me!

File: src/main/java/javacake/quiz/Question.java (hunk 2 above)
Comment (claysmilesoil) [reply src/main/java/javacake/quiz/Question.java]: implemented your suggestion, thanks!

**Pull request merged**